kew.js

```diff
@@ -839,22 +839,39 @@ function bindPromise(fn, scope, var_args) {
   }
 }
 
+
+/**
+ * Provide a more idiomatic way to immediately return a promise and ensure all
+ * subsequent code (e.g., top-level `throw`s) is promisified.
+ *
+ * @param {function(...)} fn
+ * @param {Object=} opt_thisArg
+ * @return {!Promise}
+ */
+function wrap(fn, opt_thisArg) {
```

Comment:
are you looking for `Q.fcall` / `Q.try`?
https://github.com/Medium/kew#fcall-for-delaying-a-function-invocation-until-the-next-tick
https://github.com/kriskowal/q/wiki/API-Reference#promisefcallargs